setup.go

```
@@ -165,11 +173,45 @@ func setup(c *caddy.Controller) error {
 		}
 	}
 
-	fp.httpTransport.DialContext = (&net.Dialer{
-		Timeout:   fp.dialTimeout,
-		KeepAlive: 30 * time.Second,
-		DualStack: true,
-	}).DialContext
+	if fp.upstream != "" {
+		fixedURL, err := url.Parse(fp.upstream)
+		if err != nil {
+			return c.ArgErr()
```

Better to return an actual error from parsing